### PR TITLE
Implement unspecified websocket close code (fixes issue #8158)

### LINKED ISSUE #8158: Implement unspecified websocket close code
`tests/wpt/web-platform-tests/websockets/Secure-Close-1005-verify-code.htm` is a test that closes the websocket connection without a specified code (ie. the optional first argument is missing). Our implementation chooses not to send a status code in this case, but [the specification](https://tools.ietf.org/html/rfc6455#section-7.1.5) says that we should default to 1005 in this case in `WebSocket::Close`. This change should make at least one test pass when running `./mach test-wpt test/wpt/web-platform-tests/websockets`, so the steps for "Updating test expectations" in [the README](http://mxr.mozilla.org/servo/source/tests/wpt/README.md) should be followed.

Code: components/script/dom/websocket.rs

### PR DESCRIPTION
**Pull request by yanirs**
Fixes #8158.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8693)

### CODE REVIEW

File: tests/wpt/metadata/websockets/Secure-Close-1005-verify-code.htm.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [Secure-Close-1005-verify-code.htm]
   type: testharness
   [W3C WebSocket API - Create Secure WebSocket - Close the Connection - close() - return close code is 1005 - Connection should be closed]
-    expected: FAIL
```
Comment (dzbarsky) [inline tests/wpt/metadata/websockets/Secure-Close-1005-verify-code.htm.ini]: Just delete this file.

**Comment by jdm** [comment]
@bors-servo: r+
Looks good! Thanks for doing this :)

**Pull request merged**